espnet2/asr/frontend/asteroid_frontend.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+#  2020, Technische UniversitÃ¤t MÃ¼nchen;  Ludwig KÃ¼rzinger
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+"""Sliding Window for raw audio input data."""
+
+from typing import Tuple
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from asteroid_filterbanks import Encoder, ParamSincFB
+from typeguard import check_argument_types
+
+from espnet2.asr.frontend.abs_frontend import AbsFrontend
+
+
+class AsteroidFrontend(AbsFrontend):
+    """Asteroid Filterbank Frontend.
```

Comment:
Since Asteroid actually provides various encoders, how about renaming this to `SincFrontend` to avoid confusion?

Reply:
Not so sure.. the actual name of the layer is "analytic filterbank layer" in M. Pariente, S. Cornell, A. Deleforge and E. Vincent, “Filterbank design for end-to-end speech separation,” in Proc. ICASSP, 2020. But, as we discussed once offline, "analytic filterbank layer" can be a broad term.

I added some explanation for this comment in the file :)